File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        buckets = [0] * (n + 1)
        
        for elem in citations:
            buckets[min(elem, n)] += 1
        
        accum = list(accumulate(buckets[1:][::-1]))[::-1]  
        compar = [accum[i] >= i + 1 for i in range(n)]  
        return (compar + [0]).index(0)   

File: 0274-h-index/0274-h-index.ts
function hIndex(citations: number[]): number {
    const n = citations.length;
    const buckets: number[] = new Array(n + 1).fill(0);

    for (const elem of citations) {
        buckets[Math.min(elem, n)] += 1;
    }

    const accum: number[] = [...buckets.slice(1)].reverse().reduce((acc, val) => {
        acc.push((acc.length > 0 ? acc[acc.length - 1] : 0) + val);
        return acc;
    }, []).reverse();

    const compar: boolean[] = Array.from({ length: n }, (_, i) => accum[i] >= i + 1);

    return compar.concat([false]).indexOf(false);
};